### PR TITLE
Feature/admin 202 banking validation

### PR DESCRIPTION
**Pull request by dabick14**

### CODE REVIEW

File: api/src/resolvers/banking/banking-cypher.ts [hunk 1]
```diff
@@ -54,3 +54,26 @@ DELETE r
 
 RETURN record, church.name AS churchName, date.date AS date
 `
+export const lastButOneServiceRecord = `
+MATCH (record:ServiceRecord {id: $serviceRecordId})
+MATCH (record)<-[:HAS_SERVICE]-(:HistoryLog)<-[:HAS_HISTORY]-(fellowship) 
+WHERE fellowship:Fellowship OR fellowship:ClosedFellowship
```
Comment (jaedag) [inline api/src/resolvers/banking/banking-cypher.ts:60]: Is there a need for searching closed fellowship?

File: api/src/resolvers/banking/banking-cypher.ts [hunk 2]
```diff
@@ -54,3 +54,26 @@ DELETE r
 
 RETURN record, church.name AS churchName, date.date AS date
 `
+export const lastButOneServiceRecord = `
```
Comment (jaedag) [inline api/src/resolvers/banking/banking-cypher.ts:57]: How many DB hits on this??

File: api/src/resolvers/banking/banking-resolver.ts [hunk 3]
```diff
@@ -26,6 +54,8 @@ const bankingMutation = {
 
     const { merchantId, auth } = getStreamFinancials(args.stream_name)
 
+    await checkIfLastServiceBanked(args.serviceRecordId, context)
+
     // This code checks if there has already been a successful transaction
     const transactionResponse = rearrangeCypherObject(
```
Comment (jaedag) [inline api/src/resolvers/banking/banking-resolver.ts:60]: I think we should check if the last offering was banked after checking if there has already been a successful transaction. 

It fees more efficient that way

File: api/src/resolvers/banking/banking-cypher.ts [hunk 4]
```diff
@@ -54,3 +54,26 @@ DELETE r
...
+MATCH (record:ServiceRecord {id: $serviceRecordId})
+MATCH (record)<-[:HAS_SERVICE]-(:HistoryLog)<-[:HAS_HISTORY]-(fellowship) 
+WHERE fellowship:Fellowship OR fellowship:ClosedFellowship
+WITH fellowship
+MATCH (record:ServiceRecord)<-[:HAS_SERVICE]-(:HistoryLog)<-[:HAS_HISTORY]-(fellowship)
```
Comment (jaedag) [inline api/src/resolvers/banking/banking-cypher.ts:62]: This should be :ServiceLog not :HistoryLog

**Review by jaedag — CHANGES_REQUESTED** [review]
Please address comments and resolve

File: api/src/resolvers/banking/banking-cypher.ts (hunk 2 above)
Comment (dabick14) [reply api/src/resolvers/banking/banking-cypher.ts:57]: When there's an index on serviceRecord id, it's less that 200 hits. Without the index it's in the thousands

File: api/src/resolvers/banking/banking-cypher.ts (hunk 1 above)
Comment (dabick14) [reply api/src/resolvers/banking/banking-cypher.ts:60]: If the closed fellowships show up on the church list for banking, there is a need.

File: api/src/resolvers/banking/banking-cypher.ts (hunk 2 above)
Comment (dabick14) [reply api/src/resolvers/banking/banking-cypher.ts:57]: With the Index - 175

File: api/src/resolvers/banking/banking-resolver.ts (hunk 3 above)
Comment (dabick14) [reply api/src/resolvers/banking/banking-resolver.ts:60]: Ok. done

File: api/src/resolvers/banking/banking-cypher.ts (hunk 4 above)
Comment (dabick14) [reply api/src/resolvers/banking/banking-cypher.ts:62]: Ok. Done

File: api/src/resolvers/banking/banking-resolver.ts [hunk 5]
```diff
@@ -172,6 +202,25 @@ const bankingMutation = {
       },
     }
   },
+  SubmitBankingSlip: async (object: any, args: any, context: Context) => {
```
Comment (jaedag) [inline api/src/resolvers/banking/banking-resolver.ts:205]: I thought we agreed to type the arguments

**Pull request closed without merging**